### PR TITLE
Create a DEX on Avalanche updated

### PR DESCRIPTION
**Pull request by Devilla**
Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request" make sure you have read the [Tutorial Contribution Guidelines](https://app.gitbook.com/@figment-learn/s/learn-docs/other/tutorial-guidelines). If you don't follow those guidelines it might take longer for us to review.

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://app.gitbook.com/@figment-learn/s/learn-docs/other/tutorial-guidelines).

In particular:

- [x] I've followed the [template structure](Structure).
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've ran a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've ran the tutorial and the code works.

### CODE REVIEW

**Comment by pyr0gan** [comment]
@Devilla Please tag me when this is complete. If you finish it over the weekend, I can review it on Monday 🚀

**Comment by Devilla** [comment]
@pyr0gan, please update!

**Comment by yash-sharma1** [comment]
@devilla Can you please improve the introduction section. Currently it only explains what DEX is but not what all you're going to cover in the tutorial, what are it's benefits over CEX etc. 

```
A decentralized exchange is a network, where anyone can exchange cryptocurrency tokens over blockchain by executing a transaction between two respective tokens, say AVAX with LINK, DAI or DevToken.

```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L3) (Line 3)

**Comment by yash-sharma1** [comment]
Please explain what is AvaSwap and why we're using it.

```
**Create AvaSwap directory and install dependencies**

```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L16) (Line 16)

**Comment by yash-sharma1** [comment]
Can you please explain the AvaSwap contract, comment the code so others can understand and explain it functionality.

```
# Add AvaSwap.sol
```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L69) (Line 69)

**Comment by yash-sharma1** [comment]
Before conclusion can you please put all the pieces together and show what will be final UI and how it looks. Providing a GIF of the screen working with DEX would be great.

```
# Conclusion

```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L724) (Line 724)

**Comment by yash-sharma1** [comment]
For overall code - Please add comments in the code and explain all the important separately like for smart contract.

File: network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md [hunk 1]
```diff
@@ -537,13 +563,15 @@ export default BuyForm;
 
 **Sell Tokens:**
```
Comment (yash-sharma1) [inline network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md:564]: What do you mean with "Sell Tokens" do we need to add it to code somewhere or seomething else it's confusing, please make it more clear and explain the step

**Comment by yash-sharma1** [comment]
Similar to "Sell Tokens" Explain it too

```
Check AVAX balance:

```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L712) (Line 712)

**Comment by yash-sharma1** [comment]
Can you please leverage DataHub Avalanche RPC for deploying the contract and interacting with it

**Comment by yash-sharma1** [comment]
Please update all these links. All these links are broekn. Check learn.figment.io and link here. CC - @pyr0gan 

```
# Prerequisites
```
								
[network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L5) (Line 5)

File: network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md [hunk 2]
```diff
@@ -43,25 +41,37 @@ truffle init
...
 
 ```javascript
-const Web3 = require('web3');
-const protocol = "http";
-const ip = "localhost";
-const port = 9650;
+require('dotenv').config();
+const HDWalletProvider = require("@truffle/hdwallet-provider");
+
+//Account credentials from which our contract will be deployed
```
Comment (pyr0gan) [inline network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md:50]: Depending on your ESLint configuration (if you use ESLint), no whitespace before comments may generate a warning. See https://eslint.org/docs/rules/spaced-comment for details.

```
// <-- Remember to add a space before you type the comment!
```

**Review by pyr0gan — COMMENTED** [review]
> Before conclusion can you please put all the pieces together and show what will be final UI and how it looks. Providing a GIF of the screen working with DEX would be great.
> 
> ```
> # Conclusion
> ```
> 
> [network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L724) (Line 724)

👍  If you need any help with this @Devilla, let me know.

**Comment by Devilla** [comment]
> > Before conclusion can you please put all the pieces together and show what will be final UI and how it looks. Providing a GIF of the screen working with DEX would be great.
> > ```
> > # Conclusion
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md](https://github.com/figment-networks/datahub-learn/blob/481c3a77a80f32a4d68164863ecd8b632b21ef90/network-documentation/avalanche/tutorials/create-a-dex-with-avalanche.md#L724) (Line 724)
> 
>  If you need any help with this @Devilla, let me know.

@pyr0gan  Yeah, I have recorded a video will send it over to you if want to review or edit it.

**Comment by Devilla** [comment]
@yash-sharma1 @pyr0gan Updated please check.

**Review by pyr0gan — APPROVED** [review]

**Pull request merged**